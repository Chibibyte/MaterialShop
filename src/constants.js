export default {
    BUTTONICONS: {
        AUTHENTICATION: 'mdi-login-variant',
        ADDRESS: 'mdi-package-variant',
        BANKACCOUNT: 'mdi-bank',
        SHIPPING: 'mdi-truck',
        PAYMENT: 'mdi-cash',
        SUMMARY: 'mdi-text-box-check-outline',
        CHECKOUT_COMPLETE: 'mdi-check',
        ADD: 'mdi-plus',
        REMOVE: 'mdi-minus',
        CANCEL: 'mdi-window-close',
        CONFIRM: 'mdi-check',
        SHOPPING: 'mdi-shopping',
        SUCCESS: 'mdi-check',
        FAILURE: 'mdi-window-close',
        WARNING: 'mdi-alert-circle-outline'
    }
}