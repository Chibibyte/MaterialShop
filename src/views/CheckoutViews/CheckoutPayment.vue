<template>
  <CheckoutBase
    :next_disabled="!checkoutStates.payment"
    title="Payment"
    subtitle="Choose a payment method"
    next_target="summary"
    :next_icon="$constants.BUTTONICONS.SUMMARY"
    :icon="$constants.BUTTONICONS.PAYMENT"
  >
    <SelectOrCreate
      :updateContent="false"
      :contentArrName="'paymentServices'"
      chState="payment"
      :dataToKeyFunction="dataToKey"
      :emptyDataFunction="emptyData"
      :dataDialogText="{addText:'Add Payment Service', removeText:'Remove Payment Service', updateText:'Update Payment Service'}"
    >
      <template #main="slotProps">
        <Payment disabled v-model="slotProps.data"></Payment>
      </template>

      <template v-for="sName in slotNames" #[sName]="slotProps">
        <Payment
          :validate="slotProps.validate"
          :key="'payment-'+sName"
          :disabled="slotProps.disabled"
          :value="slotProps.data"
          @input="slotProps.callback"
        ></Payment>
      </template>
    </SelectOrCreate>
  </CheckoutBase>
</template>

<script>
/**
 * Business Logic
 *    -display possible payment Service to choose from
 *    -when user clicks on service
 *        -it gets selected as the active payment service
 *        -checkoutStates.payment gets set
 *
 * Inputs:
 *      -$store.state.checkout.checkoutStates
 * Outputs:
 */
import CheckoutBase from "@/views/CheckoutViews/CheckoutBase.vue";
import Payment from "@/components/Payment.vue";
import SelectOrCreate from "@/components/SelectOrCreate.vue";
export default {
  name: "CheckoutPayment",
  components: {
    Payment,
    CheckoutBase,
    SelectOrCreate
  },
  data() {
    return {
      checkoutStates: this.$store.state.checkout.checkoutStates,
      slotNames: ["add", "remove", "update"]
    };
  },
  methods: {
    emptyData() {
      return {
        name: ""
      };
    },
    dataToKey(payment) {
      return payment.name;
    }
  }
};
</script>