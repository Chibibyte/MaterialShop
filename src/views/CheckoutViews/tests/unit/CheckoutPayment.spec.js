
import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils';
import CheckoutPayment from '@/views/CheckoutViews/CheckoutPayment.vue';
import constants from '@/constants.js'

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.prototype.$constants = constants;

let store;
let state;
let modules;
let stubs = [
    'v-container',
    'v-row',
    'v-col'
]
let wrapper;

beforeEach(() => {
    state = {}
    modules = {
        noncheckout: {
            namespaced: true,
            state() {
                return {
                }
            }
        },
        checkout: {
            namespaced: true,
            state() {
                return {
                    checkoutStates: {
                        authentication: false,
                        deliveryAddress: false,
                        bankaccount: false,
                        shipping: false,
                        payment: false,
                        summary: false,
                    }
                }
            }
        }
    }
    store = new Vuex.Store({
        state,
        modules
    })
    wrapper = shallowMount(CheckoutPayment, { stubs, store, localVue });
})

// initial
describe('Component', () => {
    test('is instantiated', () => {
        expect(wrapper.isVueInstance).toBeTruthy();
    });

    test('default snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    })
});



