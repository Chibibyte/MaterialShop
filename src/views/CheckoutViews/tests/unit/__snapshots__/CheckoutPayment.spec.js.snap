// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component default snapshot 1`] = `
<checkoutbase-stub
  icon="mdi-cash"
  next_disabled="true"
  next_icon="mdi-text-box-check-outline"
  next_name="next"
  next_target="summary"
  subtitle="Choose a payment method"
  title="Payment"
>
  <selectorcreate-stub
    chstate="payment"
    contentarrname="paymentServices"
    datadialogtext="[object Object]"
    datatokeyfunction="function () { [native code] }"
    emptydatafunction="function () { [native code] }"
  />
</checkoutbase-stub>
`;
